features/all-positions/PositionActionsDialog.tsx

```diff
@@ -0,0 +1,583 @@
+        ? Math.max(0, Math.floor(pendingTransferTimeRemaining / 3600)) +
+          ":" +
+          Math.max(0, Math.floor((pendingTransferTimeRemaining % 3600) / 60)) +
+          ":" +
+          Math.max(0, (pendingTransferTimeRemaining % 3600) % 60)
+        : "None";
+
+    const underCollateralizedPrice =
+      Number(sponsorPosition.collateral) /
+      (Number(sponsorPosition.tokensOutstanding) *
+        parseFloat(fromWei(collReq)));
+
+    const underCollateralizedPercent =
+      ((Number(latestPrice) - Number(underCollateralizedPrice)) /
+        Number(latestPrice)) *
+      100;
+
+    const collBalanceTooLow = () => {
+      if (tabIndex == "deposit") {
+        return collBalanceNum < collateralToDepositNum;
+      }
+      if (tabIndex == "liquidate") {
+        return collBalanceNum < finalFeeNum;
+      }
+    };
+
+    const tokenBalanceTooLow = tokenBalanceNum < maxTokensToLiquidateNum;
+
+    const needCollateralAllowance = () => {
+      if (tabIndex == "deposit") {
+        if (collAllowance === null || collateralToDeposit === null) return true;
+        if (collAllowance === "Infinity") return false;
+        return collAllowance < parseFloat(collateralToDeposit);
+      }
+      if (tabIndex == "liquidate") {
+        if (collAllowance === null || finalFee == null) return true;
+        if (collAllowance === "Infinity") return false;
+        return collAllowance < finalFee;
+      }
+      return true;
+    };
+    const needTokenAllowance = () => {
+      if (tokenAllowance === null || maxTokensToLiquidate === null) return true;
+      if (tokenAllowance === "Infinity") return false;
+      return tokenAllowance < parseFloat(maxTokensToLiquidate);
+    };
+
+    // Liquidation requires both collateral and synthetic to be approved. This
+    // results in a number of different configurations the user could be in depending
+    // on their interaction with this Dapp. Process the text here to keep the
+    // component simple.
+    const liquidationNeedAllowanceText = () => {
+      if (!needCollateralAllowance() && !needTokenAllowance()) {
+        return null;
+      }
+      if (needCollateralAllowance() && !needTokenAllowance()) {
+        return `You will need to sign two transactions one to approve collateral ${collSymbol} and a second to preform the liquidation.`;
+      }
+      if (!needCollateralAllowance() && needTokenAllowance()) {
+        return `You will need to sign two transactions one to approve synthetic ${tokenSymbol} and a second to preform the liquidation.`;
+      }
+      if (needCollateralAllowance() && needTokenAllowance()) {
+        return `You will need to sign three transactions one to approve synthetic ${tokenSymbol}, one to approve collateral ${collSymbol} and a third to preform the liquidation.`;
+      }
+    };
+
+    const executeDeposit = async () => {
+      if (collateralToDeposit && !collBalanceTooLow() && emp) {
```

Comment:
Instead of all of these `null` checks, wouldn't it be easier to just not render the entire component unless: 

```
if (emp !== null && tokenAllowance !== null && collateralBalance !== null, ...etc)` 
```

This is running into the same issue that we addressed for all of the existing features in #68.

For example, look at how `Deposit.tsx` does one massive null check at the beginning before rendering the component. This way you don't have to repeatedly check for null variables

Reply:
I already have the render check set up. I'll expand this accordingly to ensure it covers all variables used